### ISSUE TITLE
LuckyRecord::Form(T) can't create fields containing empty strings

### ISSUE THREAD

**Issue by BrucePerens** [luckyframework/lucky#425]
This code in lucky_record/src/lucky_record/form.cr refuses to create a field containing an empty string. validate_required tests for blank?, and I guess we should be testing for not nil here.

```
    def required_fields
      Tuple.new(
        {% for field in fields %}
          {% if !field[:nilable] && !field[:autogenerated] %}
            {{ field[:name] }},
          {% end %}
        {% end %}
      )
    end

    def after_prepare
      validate_required *required_fields
    end


```

**Comment by paulcsmith** [comment]
@BrucePerens I'm not sure I understand the problem. Can you paste some code from your form and the params you are expecting to work? Any info you can provide would be wonderful

**Comment by paulcsmith** [comment]
Also, is the error a compile time error, or a run time error?

**Comment by BrucePerens** [comment]
Run time. This is in my UserForm#prepare
self.reset_token.value = ""
That breaks, complaining that reset_token is required. The following does not break.
self.reset_token.value = "none"

**Comment by paulcsmith** [comment]
Yes, that is correct, but I don't know if it is good/bad semantics. I think for web applications, this behavior makes a lot of sense, and it is also how Rails, Phoenix and other frameworks treat presence/required validations. Here are some examples to illustrate why this behavior was chosen:

Let's say we allowed `""` to be saved. And we have this situation:

* You have a User model with a name column that are marked as required.
* Someone leaves their email and name blank
* The parameters would then be two blank strings: `""`
* The form would save successfully since technically the values are set

The problem, of course, is that this isn't what we want. We want *something* to be in the email and name column. We could add another validation for validating the size of the column be at least 1, but what if they left an empty space? We don't want that either

Allowing empty strings also makes the type system less helpful in most cases. Let's take the `reset_token` as an example:

If Lucky allowed `""` for required columns You might do something like this by accident and the compiler would be fine with it:

```crystal
PasswordResetEmail.new(token: user.reset_token).deliver

class PasswordResetEmail < BaseEmail
  def initialize(@token : String)
  end
end
```

This would work and the compiler would not warn you that the token might be useless. It would happily let the empty `""` in and send an email with an empty token. By disallowing empty `""` and instead using an optional column, you'd get an error at compile time that makes sure you check if the `reset_token` is there.

I hope that makes sense and helps illustrate why `""` are treated in this way and optional columns are recommended when you want to allow `""`
